pkg/querier/remote_read_test.go

```diff
@@ -91,25 +131,168 @@ func TestRemoteReadHandler(t *testing.T) {
 	require.Equal(t, expected, response)
 }
 
-type mockQuerier struct {
-	matrix model.Matrix
-}
-
-func (m mockQuerier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Matcher) storage.SeriesSet {
-	if sp == nil {
-		panic(fmt.Errorf("select params must be set"))
+func TestStreamedRemoteRead(t *testing.T) {
+	tcs := map[string]struct {
+		samples         []model.SamplePair
+		expectedResults []*client.StreamReadResponse
+	}{
+		"with 120 samples, we expect 1 frame with 1 chunk": {
+			samples: getNSamples(120),
+			expectedResults: []*client.StreamReadResponse{
+				{
+					ChunkedSeries: []*client.StreamChunkedSeries{
+						{
+							Labels: []mimirpb.LabelAdapter{{Name: "foo", Value: "bar"}},
+							Chunks: []client.StreamChunk{
+								{
+									MinTimeMs: 0,
+									MaxTimeMs: 119,
+									Type:      client.XOR,
+									Data:      []byte{0x0, 0x78, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0xc4, 0x57, 0xfe, 0xc2, 0x5f, 0xff, 0x6c, 0x6, 0xd6, 0x16, 0xda, 0xd, 0xb0, 0x2d, 0x2d, 0x42, 0x7b, 0x70, 0x36, 0xd0, 0xb4, 0xb6, 0x7, 0xd1, 0x4d, 0x16, 0xb1, 0x3d, 0xbc, 0x1b, 0x70, 0x5a, 0x5b, 0x43, 0xe8, 0xa6, 0x8b, 0x60, 0x9e, 0x85, 0x1a, 0x14, 0xe8, 0x51, 0xa1, 0x69, 0x1f, 0xfb, 0x80, 0x36, 0xf0, 0xb4, 0xb7, 0x7, 0xd1, 0x4d, 0x16, 0xd1, 0x3d, 0xa, 0x34, 0x29, 0xd0, 0xa3, 0x42, 0xd8, 0x2f, 0xd0, 0x50, 0xd0, 0x51, 0xd0, 0x50, 0xd0, 0x53, 0xd0, 0x50, 0xd0, 0x51, 0xd0, 0x50, 0xd0, 0x5a, 0xc6, 0xfd, 0xc4, 0x1b, 0x80, 0x5a, 0x5b, 0xc3, 0xe8, 0xa6, 0x8b, 0x70, 0x9e, 0x85, 0x1a, 0x14, 0xe8, 0x51, 0xa1, 0x6d, 0x17, 0xe8, 0x28, 0x68, 0x28, 0xe8, 0x28, 0x68, 0x29, 0xe8, 0x28, 0x68, 0x28, 0xe8, 0x28, 0x68, 0x2d, 0x83, 0x7e, 0x81, 0x41, 0xa0, 0x50, 0xe8, 0x14, 0x1a, 0x5, 0x1e, 0x81, 0x41, 0xa0, 0x50, 0xe8, 0x14, 0x1a, 0x5, 0x3e, 0x81, 0x41, 0xa0, 0x50, 0xe8, 0x14, 0x1a, 0x4},
```

Comment:
would it be possible to generate these byte slices in this unit test, instead of defining them like this?
if in the future somebody wants to modify this test it will be hard to figure out how to modify this test data, if it were generated in the unit test then it would be much easier to modify it.

Reply:
otherwise, if it's not possible to generate them in the unit test, it would be nice to have a comment which explains how they were generated

Reply:
This is raw chunk. Bytes can be generated by creating new XOR chunk (`chunkenc.NewXORChunk()`), appending samples to it (via chunk's Appender()) and then calling chunks `Bytes()` method.

(Note that `NewSeriesSetToChunkSet` used by test writes max 120 samples per chunk, hence 121 samples generate two chunks, see `seriesToChunkEncoderSplit` const)

Reply:
done @ 6e557a76b3b31dfc8ff82f427dbef6de18859933